### ISSUE TITLE
yolov5s pyTorch model to onnx model

### ISSUE THREAD

**Issue by dadaligoudan** [ultralytics/yolov5#2578]
## ❔Question
I want to use models/export.py to transfer yolov5s.pt model trained to onnx model.  But using onnx model to detect objects , I find there are some false detection and missed detection compared to using pyTorch model. 
So i checked some weights ,the detect module bias of pyTorch model is :
tensor([-0.15479,  0.01307,  0.74756,  0.04785, -3.01758, -0.84033, -0.29150, -2.00195, -3.61328, -0.15271,  0.01045,  0.32007, -0.41113, -2.90625, -0.64697, -0.65723, -1.99121, -3.69531, -0.15564,  0.01717,  0.04422, -0.69482, -2.82031, -0.75439, -0.45142, -1.93848, -3.82227], device='cuda:0')
The same module bias of onnx model is:
[
    -0.15478515625,
    0.01306915283203125,
    0.74755859375,
    0.0478515625,
    -3.017578125,
    -0.84033203125,
    -0.29150390625,
    -2.001953125,
    -3.61328125,
    -0.1527099609375,
    0.0104522705078125,
    0.320068359375,
    -0.4111328125,
    -2.90625,
    -0.64697265625,
    -0.6572265625,
    -1.9912109375,
    -3.6953125,
    -0.1556396484375,
    0.0171661376953125,
    0.044219970703125,
    -0.69482421875,
    -2.8203125,
    -0.75439453125,
    -0.451416015625,
    -1.9384765625,
    -3.822265625
]
There are some small difference between them. I think it is the reason that caused the problem. Can someone tell me how i can solve this problem?

**Comment by glenn-jocher** [comment]
@dadaligoudan ONNX and PyTorch models should return near identical inference results.

I'm not sure you can compare weights directly as ONNX models are fused while the base YOLOv5 models are not. You can fuse a YOLOv5 pytorch model with model.fuse().

**Comment by dadaligoudan** [comment]
> @dadaligoudan ONNX and PyTorch models should return near identical inference results.
> 
> I'm not sure you can compare weights directly as ONNX models are fused while the base YOLOv5 models are not. You can fuse a YOLOv5 pytorch model with model.fuse().

@glenn-jocher Thanks for your reply, after using model.fuse() in export.py, i use the same nms method and conf_thres is 0.6. I  still can't get the correct detected result. There are still some false detection and missed detection.

**Comment by dadaligoudan** [comment]
> @dadaligoudan ONNX and PyTorch models should return near identical inference results.
> 
> I'm not sure you can compare weights directly as ONNX models are fused while the base YOLOv5 models are not. You can fuse a YOLOv5 pytorch model with model.fuse().

@glenn-jocher Thanks for your reply, after using model.fuse() in export.py, i use the same nms method and conf_thres is 0.6. I  still can't get the correct detected result. There are still some false detection and missed detection.

**Comment by glenn-jocher** [comment]
@dadaligoudan If you believe you have a reproducible issue, we suggest you close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/yolov5/issues/new/choose), providing screenshots and a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and diagnose your problem. Thank you!

Please note that most technical problems are due to:

- **Your modified or out-of-date code**. If your issue is not reproducible in a new `git clone` version of this repo we can not debug it. Before going further run this code and verify your issue persists:
```bash
$ git clone https://github.com/ultralytics/yolov5 yolov5_new  # clone latest
$ cd yolov5_new
$ python detect.py  # verify detection

# CODE TO REPRODUCE YOUR ISSUE HERE
```

- **Your custom data.** If your issue is not reproducible in one of our 3 common datasets ([COCO](https://github.com/ultralytics/yolov5/blob/master/data/coco.yaml), [COCO128](https://github.com/ultralytics/yolov5/blob/master/data/coco128.yaml), or [VOC](https://github.com/ultralytics/yolov5/blob/master/data/voc.yaml)) we can not debug it. Visit our [Custom Training Tutorial](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data) for guidelines on training your custom data. Examine `train_batch0.jpg` and `test_batch0.jpg` for a sanity check of your labels and images.

- **Your environment.** If your issue is not reproducible in one of the verified environments below we can not debug it. If you are running YOLOv5 locally, verify your environment meets all of the [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) dependencies specified below. If in doubt, download Python 3.8.0 from https://www.python.org/, create a new [venv](https://packaging.python.org/guides/installing-using-pip-and-virtual-environments/), and install requirements.

If none of these apply to you, we suggest you close this issue and raise a new one using the 🐛 **Bug Report template**, providing screenshots and a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) of your issue. Thank you!


## Requirements

Python 3.8 or later with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) dependencies installed, including `torch>=1.7`. To install run:
```bash
$ pip install -r requirements.txt
```


## Environments

YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Google Colab and Kaggle** notebooks with free GPU: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>


## Status

![CI CPU testing](https://github.com/ultralytics/yolov5/workflows/CI%20CPU%20testing/badge.svg)

If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are passing. These tests evaluate proper operation of basic YOLOv5 functionality, including training ([train.py](https://github.com/ultralytics/yolov5/blob/master/train.py)), testing ([test.py](https://github.com/ultralytics/yolov5/blob/master/test.py)), inference ([detect.py](https://github.com/ultralytics/yolov5/blob/master/detect.py)) and export ([export.py](https://github.com/ultralytics/yolov5/blob/master/models/export.py)) on MacOS, Windows, and Ubuntu.
